test/Coinbase/Exchange/Socket/Test.hs

```diff
@@ -97,3 +110,667 @@ waitCancelThreads action1 action2 = do
+  where time = read "2014-11-07 08:19:27.028459 UTC"
+        orderId = read "d50ec984-77a8-460a-b958-66f114b0de9b"
+
+docReceivedMarketMsg = [aesonQQ|
+  {
+      "type": "received",
+      "time": "2014-11-09T08:19:27.028459Z",
+      "product_id": "BTC-USD",
+      "sequence": 12,
+      "order_id": "dddec984-77a8-460a-b958-66f114b0de9b",
+      "funds": "3000.234",
+      "side": "buy",
+      "order_type": "market"
+  }
+  |]
+
+expectReceivedMarketMsg = ReceivedMarketMsg {msgTime = time, msgProductId = ProductId {unProductId = "BTC-USD"}, msgSequence = Sequence {unSequence = 12}, msgOrderId = OrderId {unOrderId = orderId}, msgSide = Buy, msgClientOid = Nothing, msgMarketBounds = Right (Nothing,3000.234)}
+  where time = read "2014-11-09 08:19:27.028459 UTC"
+        orderId = read "dddec984-77a8-460a-b958-66f114b0de9b"
+
+docOpenMsg = [aesonQQ|
+  {
+      "type": "open",
+      "time": "2014-11-07T08:19:27.028459Z",
+      "product_id": "BTC-USD",
+      "sequence": 10,
+      "order_id": "d50ec984-77a8-460a-b958-66f114b0de9b",
+      "price": "200.2",
+      "remaining_size": "1.00",
+      "side": "sell"
+  }
+  |]
+
+expectOpenMsg = OpenMsg {msgTime = time, msgProductId = ProductId {unProductId = "BTC-USD"}, msgSequence = Sequence {unSequence = 10}, msgOrderId = OrderId {unOrderId = orderId}, msgSide = Sell, msgRemainingSize = 1.0, msgPrice = 200.2}
+  where time = read "2014-11-07 08:19:27.028459 UTC"
+        orderId = read "d50ec984-77a8-460a-b958-66f114b0de9b"
+
+-- NOTE: Comment removed from original.
+docDoneMsg = [aesonQQ|
+  {
+      "type": "done",
+      "time": "2014-11-07T08:19:27.028459Z",
+      "product_id": "BTC-USD",
+      "sequence": 10,
+      "price": "200.2",
+      "order_id": "d50ec984-77a8-460a-b958-66f114b0de9b",
+      "reason": "filled",
+      "side": "sell",
+      "remaining_size": "0"
+  }
+  |]
+
+expectDoneMsg = DoneMsg {msgTime = time, msgProductId = ProductId {unProductId = "BTC-USD"}, msgSequence = Sequence {unSequence = 10}, msgOrderId = OrderId {unOrderId = orderId}, msgSide = Sell, msgReason = Filled, msgMaybePrice = Just 200.2, msgMaybeRemSize = Just 0.0}
+  where time = read "2014-11-07 08:19:27.028459 UTC"
+        orderId = read "d50ec984-77a8-460a-b958-66f114b0de9b"
+
+docMatchMsg = [aesonQQ|
+  {
+      "type": "match",
+      "trade_id": 10,
+      "sequence": 50,
+      "maker_order_id": "ac928c66-ca53-498f-9c13-a110027a60e8",
+      "taker_order_id": "132fb6ae-456b-4654-b4e0-d681ac05cea1",
+      "time": "2014-11-07T08:19:27.028459Z",
+      "product_id": "BTC-USD",
+      "size": "5.23512",
+      "price": "400.23",
+      "side": "sell"
+  }
+  |]
+
+expectMatchMsg = MatchMsg {msgTime = time, msgProductId = ProductId {unProductId = "BTC-USD"}, msgSequence = Sequence {unSequence = 50}, msgSide = Sell, msgTradeId = TradeId {unTradeId = 10}, msgMakerOrderId = OrderId {unOrderId = makerOrderId}, msgTakerOrderId = OrderId {unOrderId = takerOrderId}, msgSize = 5.23512, msgPrice = 400.23}
```

Comment:
Maybe shorten it
```haskell
expectMatchMsg = 
    MatchMsg
        { msgTime = time
        , msgProductId = ProductId {unProductId = "BTC-USD"}
        , msgSequence = Sequence {unSequence = 50}
        , msgSide = Sell
        , msgTradeId = TradeId {unTradeId = 10}
        , msgMakerOrderId = OrderId {unOrderId = makerOrderId}
        , msgTakerOrderId = OrderId {unOrderId = takerOrderId}
        , msgSize = 5.23512, msgPrice = 400.23
        }
```

Reply:
Sorry, I know this is nit picky and we still have to figure out our formatting tool.

Reply:
Yes, I'm still in favor of using a formatter!